service/src/main/java/bio/terra/policy/db/PaoDao.java

```diff
@@ -104,21 +107,32 @@ public void createPao(
       propagation = Propagation.REQUIRED,
       transactionManager = "tpsTransactionManager")
   public void deletePao(UUID objectId) {
-    try {
-      // Lookup the policy object
-      DbPao dbPao = getDbPao(objectId);
+    final Set<UUID> dependents = getDependentIds(objectId);
+    setPaoDeleted(objectId, true);
 
-      // Delete associated attribute set(s)
-      deleteAttributeSet(dbPao.attributeSetId());
-      deleteAttributeSet(dbPao.effectiveSetId());
+    if (dependents.isEmpty()) {
+      DbPao dbPao = getDbPao(objectId);
 
-      // Delete the policy object
-      final String sql = "DELETE FROM policy_object WHERE object_id = :object_id";
-      MapSqlParameterSource params =
-          new MapSqlParameterSource().addValue("object_id", objectId.toString());
-      tpsJdbcTemplate.update(sql, params);
-    } catch (PolicyObjectNotFoundException e) {
-      // Delete throws no error on not found
+      // First Pass: Build a map of all PAOs in our subgraph and note if they are flagged for
+      // deletion.
+      HashMap<UUID, Boolean> removablePaoMap = new HashMap<>();
```

Comment:
ha, that was how I ended up here. I was thinking of creating a new graph node with isDeletable & DbPao but it seemed strange to use a model from there. But if I move this to the graph package, then it will need some DbPao stuff and will probably need some PaoDao stuff that is private right now. Let me think on it.